### PR TITLE
feat(browser-starfish): add sample sidebar and sorting of resource summary

### PR DESCRIPTION
**Pull request by DominikB2014**
1. Adds ability to sort in resource summary page
2. Adds the existing span sample sidebar, activated when you click on a page.
3. Add mapping between resource op and UI friendly op name
image

### CODE REVIEW

File: static/app/views/performance/browser/resources/resourceTable.tsx [hunk 1]
```diff
@@ -94,7 +94,12 @@ function ResourceTable({sort}: Props) {
       return <DurationCell milliseconds={row[key]} />;
     }
     if (key === 'span.op') {
-      const opName = row[key] === 'resource.script' ? t('Javascript') : t('Image');
+      const opNameMap = {
+        'resource.script': t('Javascript'),
+        'resource.img': t('Image'),
+        'resource.iframe': t('Javascript (iframe)'),
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/resourceTable.tsx:100]: l: let's also do `resource.css`, `resource.video`, and `resource.audio`. The rest aren't important enough.

File: static/app/views/performance/browser/resources/resourceTable.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ type Row = {
   'resource.render_blocking_status': string;
   'span.description': string;
   'span.group': string;
-  'span.op': 'resource.script' | 'resource.img';
+  'span.op': `resource.${string}`;
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/resourceTable.tsx:28]: l: if we extract the type here we can make sure that `opNameMap` is typed with the correct key names.

there's also a fun TS hack you can do to get better autocomplete working here, but no need to add it.

```ts
type LiteralUnion<T extends string> = T | Omit<T, T>;

type SpanOp = `resource.${LiteralUnion<'script' | 'img' | 'css' | 'iframe'>}`
```

This will allow any string to be suffixed to `resource.X` but will fill in autocomplete when trying to look at possible resource values.

File: static/app/views/performance/browser/resources/utils/useResourcesQuery.ts [hunk 3]
```diff
@@ -53,10 +54,7 @@ export const useResourcesQuery = ({sort}: {sort: ValidSort}) => {
     'count()': row['count()'] as number,
     'spm()': row['spm()'] as number,
     'span.group': row['span.group'].toString(),
-    'resource.render_blocking_status': row['resource.render_blocking_status'] as
-      | ''
-      | 'non-blocking'
-      | 'blocking',
+    'resource.render_blocking_status': row['resource.render_blocking_status'] as string,
```
Comment (AbhiPrasad) [inline static/app/views/performance/browser/resources/utils/useResourcesQuery.ts:57]: why the change?

**Review by AbhiPrasad — COMMENTED** [review]
Nice!

**Review by AbhiPrasad — APPROVED** [review]
oh woops meant to approve - my review doesn't block

File: static/app/views/performance/browser/resources/utils/useResourcesQuery.ts (hunk 3 above)
Comment (DominikB2014) [reply static/app/views/performance/browser/resources/utils/useResourcesQuery.ts:57]: Ooops, good call, that was not intentional

File: static/app/views/performance/browser/resources/resourceTable.tsx (hunk 2 above)
Comment (AbhiPrasad) [reply static/app/views/performance/browser/resources/resourceTable.tsx:28]: hmm it worked for me in https://github.com/getsentry/sentry-javascript/blob/e507110d89fd8117bda0a83ad7f7d2ee88681615/packages/types/src/measurement.ts#L39-L41

give me a sec to play around with it.

File: static/app/views/performance/browser/resources/resourceTable.tsx (hunk 1 above)
Comment (DominikB2014) [reply static/app/views/performance/browser/resources/resourceTable.tsx:100]: done!

**Comment by DominikB2014** [comment]
> I wonder if we should display js/css logo in resource summary page next to span description. We can update https://github.com/getsentry/platformicons/ if need be, just ask for design help.

@AbhiPrasad I definitely like this idea, I think we can loop back to it once we go the module more polished.

File: static/app/views/performance/browser/resources/resourceTable.tsx (hunk 2 above)
Comment (DominikB2014) [reply static/app/views/performance/browser/resources/resourceTable.tsx]: Sounds good, I do like that idea tho, we can probs figure another way to do it in the futre.
